backend/utils/rabbitmq_client.py

```diff
@@ -0,0 +1,177 @@
+            )
+            await self.exchange.publish(
+                message,
+                routing_key=RABBITMQ_ROUTE_USER_CHAT_REPLY
+            )
+        except Exception as e:
+            logger.error(f"Error publishing message: {e}")
+
+    async def consumer_callback(
+        self,
+        message: aio_pika.IncomingMessage,
+        consumer_type: str
+    ):
+        try:
+            async with message.process():
+                body = message.body.decode()
+                reply_to = message.headers.get("reply_to")
+                ignore = message.headers.get("ignore", False)
+                if not ignore:
+                    log = f"Received {consumer_type} message: {body}"
+                    logger.info(
+                        f"{log}, Reply To: {reply_to}"
+                    )
+                # Handle the message based on reply_to value
+                if reply_to == "twiliobot":
+                    # Process message intended for TwilioBot
+                    pass
+                elif reply_to == "slackbot":
+                    # Process message intended for SlackBot
+                    pass
+        except Exception as e:
+            logger.error(f"Error processing {consumer_type} message: {e}")
+
+    async def consume_user_chats(self):
+        await self.consume(
+            RABBITMQ_QUEUE_USER_CHATS,
+            RABBITMQ_ROUTE_USER_CHAT,
+            "user chat"
+        )
+
+    async def consume_user_chat_replies(self):
+        await self.consume(
+            RABBITMQ_QUEUE_USER_CHAT_REPLIES,
+            RABBITMQ_ROUTE_USER_CHAT_REPLY,
+            "user chat reply"
+        )
+
+    async def consume_twiliobot(self):
+        await self.consume(
+            RABBITMQ_QUEUE_USER_CHAT_REPLIES,
+            "*.twiliobot",
+            "twiliobot"
+        )
+
+    async def consume(
+        self,
+        queue_name: str,
+        routing_key: str,
+        consumer_type: str
+    ):
+        try:
+            await self.connect()
+            queue = await self.channel.declare_queue(queue_name, durable=True)
+            await queue.bind(self.exchange, routing_key=routing_key)
+            await queue.consume(lambda msg: self.consumer_callback(
+                msg, consumer_type
+            ))
+        except Exception as e:
+            logger.error(f"Error consuming {consumer_type}: {e}")
+
+    async def consume_chat_history(self):
+        try:
+            await self.connect()
+            queue = await self.channel.declare_queue(exclusive=True)
+            await queue.bind(self.exchange, routing_key="#")
+            await queue.consume(
+                lambda msg: self.consumer_callback(msg, "chat history")
+            )
+        except Exception as e:
+            logger.error(f"Error consuming chat history: {e}")
+
+    async def send_magic_link(self, body: str):
+        try:
+            routing_key = f"{RABBITMQ_ROUTE_USER_CHAT_REPLY}.twiliobot"
```

Comment:
Perhaps we should have routing keys in `.env` to replace the queue names? That would make this simpler, but would it work?